tests/core/test_test.py

```diff
@@ -127,6 +128,49 @@ async def test_testing_warns_if_action_unknown(
     assert "which is not part of the training data / domain" in output
 
 
+async def test_testing_with_utilizing_retrieval_intents(
+    response_selector_agent: Agent,
+    response_selector_test_stories: Path,
+    response_selector_results: Path,
+):
+    if not response_selector_results.exists():
+        os.mkdir(response_selector_results)
+
+    result = await rasa.core.test.test(
+        stories=response_selector_test_stories,
+        agent=response_selector_agent,
+        e2e=True,
+        out_directory=response_selector_results,
+        disable_plotting=True,
+        warnings=False,
+    )
+    failed_stories_path = response_selector_results / "failed_test_stories.yml"
+    failed_stories = read_yaml(read_file(failed_stories_path, "utf-8"))
+    # check that the intent is shown correctly in the failed test stories file
+    target_intents = {
```

Comment:
optional: This test is a bit hard to read - could we just create another test file with the expected output and then compare these files?

Reply:
You mean writing these dictionaries (`target_intents` and `target_actions`) into a file and reading it during this test?

Reply:
I meant having a `"expected_failed_test_stories.yml" which you can just compare to the `failed_test_stories.yml` from these tests by comparing the strings instead of doing the workarounds you're currently doing